src/app/components/radio/radio.component.ts

```diff
@@ -0,0 +1,60 @@
+import { Component, forwardRef, Input, OnInit } from '@angular/core';
+import { ControlValueAccessor, FormBuilder, NG_VALUE_ACCESSOR, Validators } from '@angular/forms';
+
+import { radioInterface } from 'src/app/core/interface/radioInterface';
+import { requestStatus } from 'src/app/core/models/status.enum';
+
+@Component({
+  selector: 'app-radio',
+  templateUrl: './radio.component.html',
+  styleUrls: ['./radio.component.scss'],
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => RadioComponent),
+      multi: true,
+    },
+  ]
+})
+export class RadioComponent implements ControlValueAccessor, OnInit {
+
+  requestStatus = requestStatus;
+
+  @Input() list!: radioInterface;
+  disabled: boolean = false;
+
+  onChange = (value: any) => { };
+  onTouched = () => { };
+
+  constructor(public fb: FormBuilder) { }
+
+  ngOnInit(): void {
+    this.radioForm.valueChanges.subscribe((value) => {
```

Comment:
we need to unsubscribe when this component will destroy

use take until strategy https://www.digitalocean.com/community/tutorials/angular-takeuntil-rxjs-unsubscribe